LeaderboardBackend/Services/Impl/AccountConfirmationService.cs

```diff
@@ -0,0 +1,48 @@
+using LeaderboardBackend.Models.Entities;
+
+namespace LeaderboardBackend.Services;
+
+public class AccountConfirmationService : IAccountConfirmationService
+{
+    private readonly ApplicationContext _applicationContext;
+    private readonly IEmailSender _emailSender;
+
+    public AccountConfirmationService(ApplicationContext applicationContext, IEmailSender emailSender)
+    {
+        _applicationContext = applicationContext;
+        _emailSender = emailSender;
+    }
+
+    public async Task<AccountConfirmation?> GetConfirmationById(Guid id)
+    {
+        return await _applicationContext.AccountConfirmations.FindAsync(id);
+    }
+
+    public async Task<AccountConfirmation> CreateConfirmation(User user, CancellationToken token = default)
+    {
+        AccountConfirmation newConfirmation =
+            new()
+            {
+                UserId = user.Id,
+            };
+
+        _applicationContext.AccountConfirmations.Add(newConfirmation);
+
+        await _applicationContext.SaveChangesAsync(token);
+
+#pragma warning disable CS4014 // Suppress no 'await' call
+        _emailSender.EnqueueEmailAsync(
+            user.Email,
+            // TODO: Finalise the title
+            "Confirmation",
+            GenerateAccountConfirmationEmailBody(user, newConfirmation)
+        );
+#pragma warning restore CS4014
```

Comment:
I think you actually need to pay attention to this warning. If you don't use await I'm pretty sure this could run into an issue, especially because you return immediately.

Reply:
Yeah technically this can throw if the email fails to enqueue, which is something to deal with eventually. Just didn't think it was worth tackling in this PR as well.

Reply:
What should the returned error be? 401? 409?

Reply:
If the email fails to enqueue there you could argue for actually using a 500 error since the server itself has encountered an issue. You could also just swallow the error or maybe catch it and trigger a datadog alert or something.

Reply:
You can just leave it as is for now. We can worry about this later.

Reply:
I'll add a todo to handle failed email sends.

Reply:
Sounds good. Can you make it log an error too?

Reply:
We don't have a logging library yet. I've put a todo in the service file to handle that when we have a logging library installed.